hand-tracking.ts

```diff
@@ -161,97 +187,102 @@ export class HandTracking extends Component {
-
-                /* Wrist is already handled, so start at 1 */
-                for (let j = 0; j < ORDERED_JOINTS.length; ++j) {
-                    const jointName = ORDERED_JOINTS[j];
-                    const joint = this.joints[jointName];
-                    if (!joint) continue;
-
-                    let jointPose = null;
-                    const jointSpace = (inputSource.hand as unknown as XRHand).get(
-                        jointName
-                    );
-                    if (jointSpace) {
-                        jointPose = this.engine.xr.frame.getJointPose!(
-                            jointSpace,
-                            this.engine.xr.currentReferenceSpace
-                        );
-                    }
-                    if (jointPose) {
-                        this.hasPose = true;
-                        joint.resetPositionRotation();
-
-                        joint.translateLocal([
-                            jointPose.transform.position.x - invTranslation[0],
-                            jointPose.transform.position.y - invTranslation[1],
-                            jointPose.transform.position.z - invTranslation[2],
-                        ]);
-                        joint.rotateLocal(invRotation);
-                        joint.rotateObject([
-                            jointPose.transform.orientation.x,
-                            jointPose.transform.orientation.y,
-                            jointPose.transform.orientation.z,
-                            jointPose.transform.orientation.w,
-                        ]);
-
-                        if (!this.handSkin) {
-                            /* Last joint radius of each finger is null */
-                            const r = jointPose.radius || 0.007;
-                            joint.setScalingLocal([r, r, r]);
-                        }
-                    }
+        if (wristSpace) {
+            const pose = this.engine.xr!.frame!.getJointPose!(
+                wristSpace,
+                this.engine.xr!.currentReferenceSpace
+            );
+            if (pose) {
+                setXRRigidTransformLocal(this.object, pose.transform);
+            }
+        }
+
+        this.object.getRotationLocal(invRotation);
+        quat.conjugate(invRotation, invRotation);
+        this.object.getPositionLocal(invTranslation);
+
+        for (const jointName of ORDERED_JOINTS) {
+            const joint = this.joints[jointName];
+            if (!joint) continue;
+
+            const jointSpace = (inputSource.hand as unknown as XRHand).get(jointName);
+            if (!jointSpace) continue;
+
+            const jointPose = this.engine.xr!.frame!.getJointPose!(
+                jointSpace,
+                this.engine.xr!.currentReferenceSpace
+            );
+
+            if (jointPose) {
+                joint.resetPositionRotation();
+                joint.resetPositionRotation();
+
+                joint.resetPositionRotation();
```

Comment:
Why is that called 3 times?

Reply:
lol, a ctrl+v boo boo :) , will remove it